curator-recipes/src/main/java/org/apache/curator/framework/recipes/leader/LeaderLatch.java

```diff
@@ -190,6 +190,12 @@ public void close() throws IOException
         close(closeMode);
     }
 
+    // for testing
+    void closeOnDemand() throws IOException
+    {
+        internalClose(closeMode, false);
+    }
```

Comment:
For now, this is a test-only method. I'm a bit reluctant to add test-related code in production code rather than providing a utilily class instead (to be fair: we do that in curator code in other locations like the `debug*CountDownLatches` in the `LeaderLatch` implementation). I see a risk if convenient methods like this are available in production: They might make it easier to write less-consistent production code. :thinking:

Reply:
...but if you want to stick to it, I would vote for coming up with a more descriptive name for it: Something like `closeIfStillRunning`

Reply:
I think it's a design issue we can resolve in another thread. Generally, I'd prefer `close` to be idempotent but the original code isn't. Your test case can fail if:

1. Checking state != CLOSE.
2. Internally closed.
3. Outside closed.

And it happens in the CI environment once.

Your suggestion can be reasonable but I don't want to spend too much time on this bug fix PR. A package-level visible method is for internal usage :)